mcos/de_noising/de_noise.py

```diff
@@ -0,0 +1,179 @@
+from sklearn.neighbors import KernelDensity
+from scipy.optimize import minimize
+import numpy as np
+import pandas as pd
+
+
+def de_noise_covariance_matrix(covariance_matrix: np.array, n_observations: int, bandwidth: float = .25) -> np.array:
```

Comment:
Nit, but would `denoise` be cleaner than `de_noise`?

Reply:
I think it's fine either way, I went with `de_noise` since it's called `de-noising` in the paper